data/model/namespacequota.py

```diff
@@ -0,0 +1,425 @@
+import json
+
+from peewee import fn, JOIN
+
+from data.database import (
+    Repository,
+    QuotaLimits,
+    UserOrganizationQuota,
+    QuotaType,
+    Manifest,
+    get_epoch_timestamp_ms,
+    Tag,
+    RepositorySize,
+    User,
+)
+from data import model
+from data.model import (
+    organization,
+    user,
+    InvalidUsernameException,
+    notification,
+    config,
+    InvalidSystemQuotaConfig,
+)
+
+
+def verify_namespace_quota(repository_ref):
+    model.repository.get_repository_size_and_cache(repository_ref._db_id)
+    namespace_size = get_namespace_size(repository_ref.namespace_name)
+    return check_limits(repository_ref.namespace_name, namespace_size)
+
+
+def verify_namespace_quota_force_cache(repository_ref):
+    force_cache_repo_size(repository_ref)
+    namespace_size = get_namespace_size(repository_ref.namespace_name)
+    return check_limits(repository_ref.namespace_name, namespace_size)
+
+
+def verify_namespace_quota_during_upload(repository_ref):
+    size = model.repository.get_size_during_upload(repository_ref._db_id)
+    namespace_size = get_namespace_size(repository_ref.namespace_name)
+    if namespace_size is None:
+        namespace_size = 0
+
+    return check_limits(repository_ref.namespace_name, size + namespace_size)
+
+
+def check_limits(namespace_name, size):
+    limits = get_namespace_limits(namespace_name)
+    limit_bytes = 0
+    severity_level = None
+    for limit in limits:
+        if size > limit["bytes_allowed"]:
+            if limit_bytes < limit["bytes_allowed"]:
+                limit_bytes = limit["bytes_allowed"]
+                severity_level = limit["type_id"]
+
+    return {"limit_bytes": limit_bytes, "severity_level": severity_level}
+
+
+def notify_organization_admins(repository_ref, notification_kind, metadata={}):
+    if repository_ref.namespace_user.organization:
```

Comment:
You might run into the same issue here, and other places where `repository_ref` is referenced. `repository_ref` is not the type you're expecting.

So you might have to change the signature of the functions in this module to take the namespace name and repo name after all, and get the actual peewee object from `data/model/repository.py`, that way you can reference the user object associated to the repo.

Reply:
This is expecting the repository_ref object and uses the namespace_user parameter to gather the organization information if its an org.

Reply:
This is not something to fix other than explicitly defining parameter type.

Reply:
The callers of this function are using `registry_model.lookup_repository` though, which doesn't have a `namespace_user` reference. It does have a `namespace_name` reference though.

Reply:
So they're passing the repo_ref (as defined in the registry_model datatypes) and not the repo obj itself (referring to obj as peewee objects).

Reply:
Fixed.